geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreHelper.scala

```diff
@@ -0,0 +1,103 @@
+package org.locationtech.geomesa.kafka
+
+import java.util.UUID
+
+import org.locationtech.geomesa.utils.geotools.FeatureUtils
+import org.opengis.feature.simple.SimpleFeatureType
+import org.locationtech.geomesa.utils.geotools.Conversions.RichSimpleFeatureType
+
+object KafkaDataStoreHelper {
+
+  val TopicKey = "Topic"
+  val ReplayConfigKey = "ReplayConfig"
+
+  /** Creates a copy of the passed SimpleFeatureType, inserting the topic name (derived from zkPath) into user data.*/
+  def prepareForLive(sft: SimpleFeatureType, zkPath: String) : SimpleFeatureType = {
+
+    val builder = FeatureUtils.builder(sft)
+    val preparedSft = builder.buildFeatureType()
+    insertTopic(preparedSft,buildTopicName(zkPath,sft))
+    preparedSft
+  }
+
+
+  /** Creates a copy of the passed SimpleFeature type, inserting an encoded ReplayConfig into the user data. */
+  def prepareForReplay(sft: SimpleFeatureType, rConfig: ReplayConfig) : SimpleFeatureType = {
+
+    val builder = FeatureUtils.builder(sft)
+    builder.setName(buildReplayTypeName(sft.getTypeName))
+    ReplayTimeHelper.addReplayTimeAttribute(builder)
+
+    // assumes topic has been prepared, need to check though...
+    val preparedSft = builder.buildFeatureType()
+    preparedSft.getUserData.put(ReplayConfigKey,ReplayConfig.encode(rConfig))
+    preparedSft
+  }
+
+  /** modifies the passed SimpleFeature type, adding the topic string to user data */
+  def insertTopic(sft: SimpleFeatureType, topic: String) : SimpleFeatureType = {
```

Comment:
This function can return 'Unit' since it is mutating state.  (All uses I observed follow that pattern.